### PR TITLE
Fixed #32383 -- Added source map support to ManifestStaticFilesStorage

### PR DESCRIPTION
**Pull request by adamchainz**
ticket-32383

### CODE REVIEW

File: django/contrib/staticfiles/storage.py [hunk 1]
```diff
@@ -49,6 +49,9 @@ class HashedFilesMixin:
             r"""(url\(['"]{0,1}\s*(.*?)["']{0,1}\))""",
             (r"""(@import\s*["']\s*(.*?)["'])""", """@import url("%s")"""),
         )),
+        ("*.js", (
+            (r"(?m)^(//# (?-i:sourceMappingURL)=(.*))$", "//# sourceMappingURL=%s"),
+        )),
```
Comment (adamchainz) [inline django/contrib/staticfiles/storage.py:54]: this regex uses an `(?-i:...)` group to disable the case-insensitive flag around the `sourceMappingURL`. It's probably a bit defensive, but ideally we'd not force the case-insensitive flag on the patterns in `__init__`. But I can't think of a way of doing that in a bakcwards-compatible way.

**Review by felixxm — COMMENTED** [review]
@adamchainz Thanks for this patch :+1: 

I added `versionchanged` annotation and a test for case-insensitive mapping.

**Comment by felixxm** [comment]
buildbot, test this please.

**Comment by felixxm** [comment]
@adamchainz `test_js_source_map` doesn't work on Windows. Can you take a look?

**Comment by adamchainz** [comment]
Thanks for your updates.

I can't work out why it's failing - it's coming up with a different hash on windows for apparently no good reason? I checked with `hexdump` and there are no weird characters such as CRLF line endings in `source_map.js`. And the file should be opened in binary mode too.

I'm pushing a commit with

**Comment by felixxm** [comment]
@adamchainz I tried to investigate this and it's related with `HashedFilesMixin._stored_name()`, for the same value i.e. `cached/source_map.js.map` it returns different values on Linux and Windows :thinking:

**Comment by felixxm** [comment]
OK I found :confetti_ball: `file_hash()` creates a hash from a different content:
- ``b'{}\n'`` on Linux
- ``b'{}\r\n'`` on Windows

**Comment by felixxm** [comment]
Let's try with no newline at the end of file :detective:

**Comment by felixxm** [comment]
@adamchainz Fixed :partying_face:

**Comment by felixxm** [comment]
buildbot, test this please.

**Comment by adamchainz** [comment]
Great, thanks.

**Pull request merged**